.packit.yaml

```diff
@@ -111,31 +126,33 @@ jobs:
         - *tmt-cloud-resources
 
   # Test internal plugins
-  - <<: *test_base
+  - <<: *test-base
     <<: *internal
     <<: *require-full-tests
     identifier: internal-plugins
     fmf_url: "https://gitlab.cee.redhat.com/baseos-qe/tmt.git"
 
   # Test internal wow
-  - <<: *test_base
+  - <<: *test-base
     <<: *internal
     <<: *require-full-tests
     identifier: internal-wow
     fmf_url: "https://gitlab.cee.redhat.com/baseos-qe/integration_scripts.git"
     tmt_plan: '/tmt/integration/plan'
 
-  # Fedora releases
+  # Propose downstream pull requests
   - job: propose_downstream
     trigger: release
     dist_git_branches: *all-targets
 
+  # Create koji builds
   - job: koji_build
     trigger: commit
-    allowed_pr_authors: [ "packit", "psss", "lzachar" ]
-    allowed_committers: [ "packit", "psss", "lzachar" ]
+    allowed_pr_authors: ["packit", "all_committers"]
+    allowed_committers: ["packit", "all_committers"]
```

Comment:
This change is dangerous IMO. It would make it impossible to manually push commits without building in koji, e.g. for building in side-tag. My recommendation would be to leave it to the default (only `packit`) since there is little gained by adding the `allowed_*`, because if you have overwritten the packit PR, probably you are doing something that requires manual workflows.

Reply:
I believe this was actually needed if there was a problem with automation and I had to create the fedora dist-git pull request using the `packit` the command line. If only packit is allowed, then there is no way to do that step manually if something gets broken.

Also, what do you mean but "not possible to push commits"? The non-scratch koji builds are only created when the dist-git pull requests are merge. Or am I missing something?

Reply:
> I believe this was actually needed if there was a problem with automation and I had to create the fedora dist-git pull request using the `packit` the command line.

You mean that you would not be able to create the PR? That should not be the case, `packit propose-downstream` should simply create the PR with your credentials. The automation would of course not run, but you can use `/packit propose-downstream` in *any* issue to re-trigger as `packit` [^1], and it should work for PR comments as well, I think.

[^1]: https://packit.dev/docs/configuration/upstream/propose_downstream#retriggering

> Also, what do you mean but "not possible to push commits"? The non-scratch koji builds are only created when the dist-git pull requests are merge. Or am I missing something?

The issue is that it actually triggers on the `commit`, not on the PR merge event [^2], so when you set `allowed_committers`, **all** commits from users in that list will trigger a build, regardless if it's a merge or direct push. That is why the default for that is an empty list because it can easily lock up the configuration.

[^2]: https://packit.dev/docs/configuration/downstream/koji_build#supported-triggers

Reply:
Ok, I see, thanks much for the detailed clarification. I thought you suggested to remove both lines. So, do I understand it correctly that we should be good with just the following line?

```yaml
    allowed_pr_authors: [all_committers]
```

Seems that `packit` does not have to be there.

Reply:
> Ok, I see, thanks much for the detailed clarification. I thought you suggested to remove both lines. So, do I understand it correctly that we should be good with just the following line?
> 
> ```yaml
>     allowed_pr_authors: [all_committers]
> ```
> 
> Seems that `packit` does not have to be there.

Yes, at the very least `allowed_commiters` is dangerous. About `allowed_pr_authors` I think it's fine to expand as long as all committers know that a koji_build will always be triggered on PR commits. It's safer in this case because no sig group has explicit rights here. As for needing for `packit` to be there, I am not sure, but would be safer to include it since it's in the default and `packit` does not have `committer` rights

Reply:
@lachmanfrantisek, @lbarcziova, does `packit` need to be explicitly included in the list of `allowed_pr_authos`?

Reply:
> I think it's fine to expand as long as all committers know that a koji_build will always be triggered on PR commits.

The koji build should be only triggered when the dist-git pull request is merged, not for every commit in the pull request. And I would say that this is expected/fine.

Reply:
hi @psss , `packit` should be included explicitly in the list.

Reply:
> The issue is that it actually triggers on the commit, not on the PR merge event [2](https://github.com/teemtee/tmt/pull/3375#user-content-fn-2-d88b6dd7b4f29358180f584767225773), so when you set allowed_committers, all commits from users in that list will trigger a build, regardless if it's a merge or direct push. That is why the default for that is an empty list because it can easily lock up the configuration.

just to clarify, all commits, meaning direct pushes to the dist-git repository, but containing a specfile change and to the configured dist-git branches

Reply:
Thanks for the clarification! The `allowed_committers` line dropped in 5bcb8c98.